system_tests/batch_poster_test.go

```diff
@@ -120,3 +121,43 @@ func TestBatchPosterParallel(t *testing.T) {
 		Fail(t, "Unexpected zero balance")
 	}
 }
+
+func TestBatchPosterLargeTx(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	conf := arbnode.ConfigDefaultL1Test()
+	conf.Sequencer.MaxTxDataSize = 110000
+	l2info, nodeA, l2clientA, l1info, _, l1client, l1stack := createTestNodeOnL1WithConfig(t, ctx, true, conf, nil, nil)
+	defer requireClose(t, l1stack)
+	defer nodeA.StopAndWait()
+
+	l2clientB, nodeB := Create2ndNode(t, ctx, nodeA, l1stack, l1info, &l2info.ArbInitData, nil)
+	defer nodeB.StopAndWait()
+
+	data := make([]byte, 100000)
+	_, err := rand.Read(data)
+	Require(t, err)
+	faucetAddr := l2info.GetAddress("Faucet")
+	gas := l2info.TransferGas + 20000*uint64(len(data))
+	tx := l2info.PrepareTxTo("Faucet", &faucetAddr, gas, common.Big0, data)
+	err = l2clientA.SendTransaction(ctx, tx)
+	Require(t, err)
+	_, err = EnsureTxSucceeded(ctx, l2clientA, tx)
+	Require(t, err)
+
+	for i := 90; i > 0; i-- {
+		SendWaitTestTransactions(t, ctx, l1client, []*types.Transaction{
+			l1info.PrepareTx("Faucet", "User", 30000, big.NewInt(1e12), nil),
```

Comment:
need to test that all did get into the same block.. maybe just check how many batches were posted.
It might help to pass all transactions in a single array to SendWaitTestTramsactios so it won't wait between sending separate transactins.

Reply:
This is sending L1 txs, not L2, to help the inbox reader along. I copied it from the above batch poster test.

Reply:
I've gone ahead and removed this code and simplified it a lot by just using EnsureTxSucceededWithTimeout, as that's good enough it turns out. This should also verify that the message is intact because otherwise the transaction hash wouldn't exist.

Reply:
I've added an additional check for the block hash :)